lms/djangoapps/verify_student/services.py

```diff
@@ -61,121 +62,84 @@ def user_is_verified(cls, user, earliest_allowed_date=None):
-        """
-        Return a query set for all records with 'approved' state
-        that are still valid according to the earliest_allowed_date
-        value or policy settings.
-        """
-        return SoftwareSecurePhotoVerification.objects.filter(
-            status="approved",
-            created_at__gte=(earliest_allowed_date or earliest_allowed_verification_date()),
-        )
+        return (SoftwareSecurePhotoVerification.objects.filter(**filter_kwargs).exists() or
+                SSOVerification.objects.filter(**filter_kwargs).exists())
 
     @classmethod
     def verifications_for_user(cls, user):
         """
-        Return a query set for all records associated with the given user.
+        Return a list of all verifications associated with the given user.
         """
-        return SoftwareSecurePhotoVerification.objects.filter(user=user)
+        verifications = []
+        for verification in chain(SoftwareSecurePhotoVerification.objects.filter(user=user),
+                                  SSOVerification.objects.filter(user=user)):
+            verifications.append(verification)
+        return verifications
 
     @classmethod
     def get_verified_users(cls, users):
         """
-        Return the list of user ids that have non expired verifications from the given list of users.
-        """
-        return cls.verified_query().filter(user__in=users).select_related('user')
-
-    @classmethod
-    def verification_valid_or_pending(cls, user, earliest_allowed_date=None, queryset=None):
-        """
-        Check whether the user has a complete verification attempt that is
-        or *might* be good. This means that it's approved, been submitted,
-        or would have been submitted but had an non-user error when it was
-        being submitted.
-        It's basically any situation in which the user has signed off on
-        the contents of the attempt, and we have not yet received a denial.
-        This will check for the user's *initial* verification.
-
-        Arguments:
-            user:
-            earliest_allowed_date: earliest allowed date given in the
-                settings
-            queryset: If a queryset is provided, that will be used instead
-                of hitting the database.
-
-        Returns:
-            queryset: queryset of 'PhotoVerification' sorted by 'created_at' in
-            descending order.
-        """
-
-        valid_statuses = ['submitted', 'approved', 'must_retry']
-
-        if queryset is None:
-            queryset = SoftwareSecurePhotoVerification.objects.filter(user=user)
-
-        return queryset.filter(
-            status__in=valid_statuses,
-            created_at__gte=(
-                earliest_allowed_date
-                or earliest_allowed_verification_date()
-            )
-        ).order_by('-created_at')
+        Return the list of users that have non-expired verifications of either type from
```

Comment:
you are right - it should say iterator in this comment. I modified the above function because it turned out to be put in a variable that was reused but the only caller of this function is using the result once.